// @aws-sdk/client-glue/dist-es/models/errors.js
import { GlueServiceException as __BaseException } from "./GlueServiceException";
export class AccessDeniedException extends __BaseException {
    name = "AccessDeniedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
    }
}
export class AlreadyExistsException extends __BaseException {
    name = "AlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "AlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
export class EntityNotFoundException extends __BaseException {
    name = "EntityNotFoundException";
    $fault = "client";
    Message;
    FromFederationSource;
    constructor(opts) {
        super({
            name: "EntityNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, EntityNotFoundException.prototype);
        this.Message = opts.Message;
        this.FromFederationSource = opts.FromFederationSource;
    }
}
export class GlueEncryptionException extends __BaseException {
    name = "GlueEncryptionException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "GlueEncryptionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, GlueEncryptionException.prototype);
        this.Message = opts.Message;
    }
}
export class InternalServiceException extends __BaseException {
    name = "InternalServiceException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalServiceException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServiceException.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidInputException extends __BaseException {
    name = "InvalidInputException";
    $fault = "client";
    Message;
    FromFederationSource;
    constructor(opts) {
        super({
            name: "InvalidInputException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidInputException.prototype);
        this.Message = opts.Message;
        this.FromFederationSource = opts.FromFederationSource;
    }
}
export class OperationTimeoutException extends __BaseException {
    name = "OperationTimeoutException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OperationTimeoutException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OperationTimeoutException.prototype);
        this.Message = opts.Message;
    }
}
export class ResourceNumberLimitExceededException extends __BaseException {
    name = "ResourceNumberLimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceNumberLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNumberLimitExceededException.prototype);
        this.Message = opts.Message;
    }
}
export class ResourceNotReadyException extends __BaseException {
    name = "ResourceNotReadyException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceNotReadyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotReadyException.prototype);
        this.Message = opts.Message;
    }
}
export class FederationSourceException extends __BaseException {
    name = "FederationSourceException";
    $fault = "client";
    FederationSourceErrorCode;
    Message;
    constructor(opts) {
        super({
            name: "FederationSourceException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FederationSourceException.prototype);
        this.FederationSourceErrorCode = opts.FederationSourceErrorCode;
        this.Message = opts.Message;
    }
}
export class FederationSourceRetryableException extends __BaseException {
    name = "FederationSourceRetryableException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "FederationSourceRetryableException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FederationSourceRetryableException.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidStateException extends __BaseException {
    name = "InvalidStateException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidStateException.prototype);
        this.Message = opts.Message;
    }
}
export class ThrottlingException extends __BaseException {
    name = "ThrottlingException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.Message = opts.Message;
    }
}
export class IllegalSessionStateException extends __BaseException {
    name = "IllegalSessionStateException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IllegalSessionStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IllegalSessionStateException.prototype);
        this.Message = opts.Message;
    }
}
export class ConcurrentModificationException extends __BaseException {
    name = "ConcurrentModificationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConcurrentModificationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentModificationException.prototype);
        this.Message = opts.Message;
    }
}
export class FederatedResourceAlreadyExistsException extends __BaseException {
    name = "FederatedResourceAlreadyExistsException";
    $fault = "client";
    Message;
    AssociatedGlueResource;
    constructor(opts) {
        super({
            name: "FederatedResourceAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, FederatedResourceAlreadyExistsException.prototype);
        this.Message = opts.Message;
        this.AssociatedGlueResource = opts.AssociatedGlueResource;
    }
}
export class ColumnStatisticsTaskRunningException extends __BaseException {
    name = "ColumnStatisticsTaskRunningException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ColumnStatisticsTaskRunningException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ColumnStatisticsTaskRunningException.prototype);
        this.Message = opts.Message;
    }
}
export class IdempotentParameterMismatchException extends __BaseException {
    name = "IdempotentParameterMismatchException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IdempotentParameterMismatchException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IdempotentParameterMismatchException.prototype);
        this.Message = opts.Message;
    }
}
export class ValidationException extends __BaseException {
    name = "ValidationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
    }
}
export class ConflictException extends __BaseException {
    name = "ConflictException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
}
export class IntegrationConflictOperationFault extends __BaseException {
    name = "IntegrationConflictOperationFault";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IntegrationConflictOperationFault",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IntegrationConflictOperationFault.prototype);
        this.Message = opts.Message;
    }
}
export class IntegrationQuotaExceededFault extends __BaseException {
    name = "IntegrationQuotaExceededFault";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IntegrationQuotaExceededFault",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IntegrationQuotaExceededFault.prototype);
        this.Message = opts.Message;
    }
}
export class InternalServerException extends __BaseException {
    name = "InternalServerException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
export class KMSKeyNotAccessibleFault extends __BaseException {
    name = "KMSKeyNotAccessibleFault";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "KMSKeyNotAccessibleFault",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KMSKeyNotAccessibleFault.prototype);
        this.Message = opts.Message;
    }
}
export class ResourceNotFoundException extends __BaseException {
    name = "ResourceNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
export class OperationNotSupportedException extends __BaseException {
    name = "OperationNotSupportedException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "OperationNotSupportedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OperationNotSupportedException.prototype);
        this.Message = opts.Message;
    }
}
export class CrawlerRunningException extends __BaseException {
    name = "CrawlerRunningException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CrawlerRunningException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CrawlerRunningException.prototype);
        this.Message = opts.Message;
    }
}
export class SchedulerTransitioningException extends __BaseException {
    name = "SchedulerTransitioningException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "SchedulerTransitioningException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SchedulerTransitioningException.prototype);
        this.Message = opts.Message;
    }
}
export class IntegrationNotFoundFault extends __BaseException {
    name = "IntegrationNotFoundFault";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IntegrationNotFoundFault",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IntegrationNotFoundFault.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidIntegrationStateFault extends __BaseException {
    name = "InvalidIntegrationStateFault";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "InvalidIntegrationStateFault",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidIntegrationStateFault.prototype);
        this.Message = opts.Message;
    }
}
export class ConditionCheckFailureException extends __BaseException {
    name = "ConditionCheckFailureException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConditionCheckFailureException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConditionCheckFailureException.prototype);
        this.Message = opts.Message;
    }
}
export class TargetResourceNotFound extends __BaseException {
    name = "TargetResourceNotFound";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TargetResourceNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TargetResourceNotFound.prototype);
        this.Message = opts.Message;
    }
}
export class PermissionTypeMismatchException extends __BaseException {
    name = "PermissionTypeMismatchException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "PermissionTypeMismatchException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PermissionTypeMismatchException.prototype);
        this.Message = opts.Message;
    }
}
export class ConcurrentRunsExceededException extends __BaseException {
    name = "ConcurrentRunsExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConcurrentRunsExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentRunsExceededException.prototype);
        this.Message = opts.Message;
    }
}
export class IllegalWorkflowStateException extends __BaseException {
    name = "IllegalWorkflowStateException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IllegalWorkflowStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IllegalWorkflowStateException.prototype);
        this.Message = opts.Message;
    }
}
export class IllegalBlueprintStateException extends __BaseException {
    name = "IllegalBlueprintStateException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "IllegalBlueprintStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IllegalBlueprintStateException.prototype);
        this.Message = opts.Message;
    }
}
export class NoScheduleException extends __BaseException {
    name = "NoScheduleException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoScheduleException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoScheduleException.prototype);
        this.Message = opts.Message;
    }
}
export class SchedulerRunningException extends __BaseException {
    name = "SchedulerRunningException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "SchedulerRunningException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SchedulerRunningException.prototype);
        this.Message = opts.Message;
    }
}
export class MLTransformNotReadyException extends __BaseException {
    name = "MLTransformNotReadyException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "MLTransformNotReadyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MLTransformNotReadyException.prototype);
        this.Message = opts.Message;
    }
}
export class ColumnStatisticsTaskNotRunningException extends __BaseException {
    name = "ColumnStatisticsTaskNotRunningException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ColumnStatisticsTaskNotRunningException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ColumnStatisticsTaskNotRunningException.prototype);
        this.Message = opts.Message;
    }
}
export class ColumnStatisticsTaskStoppingException extends __BaseException {
    name = "ColumnStatisticsTaskStoppingException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ColumnStatisticsTaskStoppingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ColumnStatisticsTaskStoppingException.prototype);
        this.Message = opts.Message;
    }
}
export class CrawlerNotRunningException extends __BaseException {
    name = "CrawlerNotRunningException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CrawlerNotRunningException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CrawlerNotRunningException.prototype);
        this.Message = opts.Message;
    }
}
export class CrawlerStoppingException extends __BaseException {
    name = "CrawlerStoppingException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CrawlerStoppingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CrawlerStoppingException.prototype);
        this.Message = opts.Message;
    }
}
export class SchedulerNotRunningException extends __BaseException {
    name = "SchedulerNotRunningException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "SchedulerNotRunningException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SchedulerNotRunningException.prototype);
        this.Message = opts.Message;
    }
}
export class VersionMismatchException extends __BaseException {
    name = "VersionMismatchException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "VersionMismatchException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, VersionMismatchException.prototype);
        this.Message = opts.Message;
    }
}
